services/explorer/src/api/explorer-api/fetchDecodedTransactions.ts

```diff
@@ -0,0 +1,48 @@
+import axios, { AxiosResponse } from 'axios';
+import { useQuery } from 'react-query';
+import { CallData } from '../../models';
+import { TransactionType } from '../../utils';
+
+const getDecodedTrans = (endpoint: string, hash: string) => {
+  if (!hash || !endpoint) {
+    return;
+  }
+  const url = `${endpoint}/transaction/${hash}`;
+  return axios.get(url);
+};
+
+export const useGetDecodedTransactionsByTxHash = (
+  endpoint: string,
+  txType: TransactionType,
+  hash: string
+): {
+  data: AxiosResponse<any>;
+  isError: boolean;
+  error: unknown;
+  isLoading: boolean;
+  isFetching: boolean;
+} => {
+  const { data, isError, error, isLoading, isFetching } = useQuery(
+    ['decoded-transactions', hash, endpoint, txType],

```

Comment:
Don't need the txType here. There will not be any other tx with a different tx type.

Reply:
Right, since the components are conditionally rendered. However, I'll add a JsDoc entry to the endpoint call to let others know that this endpoint will return 404 if the tx is not available in the indexer.